### PR TITLE
LPS-122449 Remove usages of dom.delegate

### PR DESCRIPTION
**Pull request by kresimir-coko**
I've started removing usages of `dom.delegate`. 

In the meanwhile I realised that the implementation of `delegate` isn't fully functional, at least from my investigations, so I've pushed a commit that extends the utility to find an ancestor that matches the provided `selector`. 

I've also provided some examples of usage with the pattern tested working, except a couple of them - ones that have selectors for html tags or children, e.g. `'.someClassName a'`

### CODE REVIEW

**Comment by wincent** [comment]
> Another thing I discovered is that the usage commonly has `event.delegateTarget` and `.removeListener()` which I removed from a couple usages so let me know if I did it properly, I managed to test both and they seem to be working fine.

[MDN's page on `Event.srcElement`](https://developer.mozilla.org/en-US/docs/Web/API/Event/srcElement)

> Deprecated
>
> This feature is no longer recommended. Though some browsers might still support it, it may have already been removed from the relevant web standards, may be in the process of being dropped, or may only be kept for compatibility purposes. Avoid using it, and update existing code if possible; see the compatibility table at the bottom of this page to guide your decision. Be aware that this feature may cease to work at any time.

**Comment by kresimir-coko** [comment]
I've pushed another batch of changes, and updated the list of those I didn't manage to test due to not being able to find them in the UI.

I've hit a problem with one of them, [upload_multiple_file_entries_resources.jsp](https://github.com/liferay/liferay-portal/blob/master/modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/upload_multiple_file_entries_resources.jsp#L219), where after adding delegate, and doing the thing that triggers the event the page breaks, and it cannot find `Liferay`, `AUI`, here's how it looks.
image

**Comment by kresimir-coko** [comment]
I've pushed more changes here, but there's a blocker that I explained in [Slack](https://liferay.slack.com/archives/G3JBR21HA/p1604490233296300).

**Comment by kresimir-coko** [comment]
I've pushed a couple commits, replacing old ones. One of the commits expands the `delegate` utility, whereas the other commit contains some of the replacements in JSPs, let me know if I missed something.

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js [hunk 1]
```diff
@@ -35,11 +35,16 @@ function isDisabled(node) {
...
+		const {target, defaultPrevented} = event;
+
+		if (eventName === 'click' && isDisabled(target)) {
 			return;
 		}
 
-		if (!event.defaultPrevented && event.target.matches(selector)) {
+		if (
+			!defaultPrevented &&
+			(target.matches(selector) || target.closest(selector))
```
Comment (bryceosterhaus) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js:46]: @jbalsas @matuzalemsteles @wincent 

Do any of you know where this logic is handled within `metal-dom`? I was looking through the implementation of delegate there, and can't seem to figure out how they achieve this behavior without calling `closest`. Code I was looking at is [here](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L928)

I also was using [this codesandbox](https://codesandbox.io/s/trusting-gates-17le0?file=/index.html) to test

Comment (bryceosterhaus) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js:46]: calling `.closest` may be the best way to achieve this, but I didn't want to just add that without seeing how metal-dom was doing it

Comment (wincent) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js:46]: Is it coming via?

- [`isAbleToInteractWith()`](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L286)
- which [calls](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L302) `parent()`
- which [itself calls `closest()`](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L540-L542)

(`isAbletoInteractWith()` is called from [`triggerDelegatedListeners_()`](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L746)).

Based on a quick scan, so don't take my word for it...

Comment (bryceosterhaus) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js:46]: ah, I think I may just be blind. Your comment made me re-look and noticed this line immediately https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L756

In that case, I think calling `.closest` should be fine here.

@kresimir-coko can you add some unit tests to account for this use case?

Comment (kresimir-coko) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js:46]: Yes, I can add some unit tests, of course.

@jbalsas should I split this into 2 tasks on JIRA, one for the improvements to the implementations, and the other for usages?

Comment (jbalsas) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js:46]: > @jbalsas should I split this into 2 tasks on JIRA, one for the improvements to the implementations, and the other for usages?

There's already a task in JIRA for that. Just use that LPS number

**Comment by kresimir-coko** [comment]
I've pushed new commits, removing the old ones, the commits are split into 2, one for JSPs and one for JS files.

I've tested a couple of patterns used in JSPs and one (and only) pattern used in JS files.

**Comment by wincent** [comment]
GitHub is showing conflicts on this one @kresimir-coko, so looks like it needs a rebase.

**Comment by kresimir-coko** [comment]
I see @wincent thanks for the ping, I'll take a look.

**Comment by kresimir-coko** [comment]
Hey @wincent I've pushed commits with removed conflicts!

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
Hey @wincent I'm trying to see which files/modules have SF errors in them, but I don't know the login details for the test platform:
image

**Comment by jbalsas** [comment]
Hey @kresimir-coko, just expand the comment above:

Screen Shot 2020-11-10 at 13 54 40

And locally run `ant format-source` from the `portal-impl` folder to see the full scan.

**Comment by kresimir-coko** [comment]
> And locally run ant format-source from the portal-impl folder to see the full scan.

Didn't know we had this going for us, awesome, ran it and pushed a commit fixing those SF problems.

**Comment by wincent** [comment]
> Hey @wincent I'm trying to see which files/modules have SF errors in them, but I don't know the login details for the test platform:

At least some of them can be seen in the GitHub comment:

```
     [exec] apps/journal/journal-content-web/src/main/resources/META-INF/resources/configuration.jsp: BAD
     [exec] apps/journal/journal-web/src/main/resources/META-INF/resources/view_more_menu_items.jsp: BAD
     [exec] apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/common/history.jsp: BAD
     [exec] apps/layout/layout-admin-web/src/main/resources/META-INF/resources/select_master_layout.jsp: BAD
     [exec] apps/layout/layout-admin-web/src/main/resources/META-INF/resources/select_style_book.jsp: BAD
     [exec] apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/page.jsp: BAD
     [exec] apps/product-navigation/product-navigation-product-menu-web/src/main/resources/META-INF/resources/portlet/pages_tree.jsp: BAD
     [exec] apps/wiki/wiki-web/src/main/resources/META-INF/resources/wiki/page_iterator.jsp: BAD
     [exec] Prettier checked 54 files, 13 files have problems
```

but in any case, the login details are you Liferay email address name (ie. up to but not including the `@`) and your Liferay email password.

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
(BTW @kresimir-coko you should always feel free to run those `ci:*` commands on your own PRs whenever you update them, so as to reduce latency).

**Comment by kresimir-coko** [comment]
This is my first time actually having to open Jenkins and find where stuff is breaking, so 🐻  with me a bit.

I think I found one instance of failing tests, this [Jenkins report](https://test-1-14.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/457406/AXIS_VARIABLE=20,label_exp=!master/testReport/) should correspond to [this file](https://github.com/liferay/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/view.jsp#L93). 

Am I looking at the right place inside Jenkins, and is there  a way to get a list of breaking files when looking at these failed tests, so that I don't have to scour failed tests in Jenkins and find a file that matches some keywords in Jenkins?

**Comment by wincent** [comment]
I generally don't scour anything in Jenkins, especially for Poshi errors, because the stuff you see in the Jenkins console doesn't provide enough context:

```
com.liferay.poshi.runner.exception.ElementNotFoundPoshiRunnerException: Element is not present at "//a[contains(@class,'tab') and .='Elements'] | //a[contains(@class,'nav-link active') and .='Elements']"
```

ie. why didn't it find the element? Is it just not there? Did the page time-out? Dit the class name change? Did some other JS error happen and things blew up etc? And how did you you find this? Are you sure it's not one of the famous "failures in common with upstream"? If you can view this in the Poshi interface you might be able to figure out what is happening (because it contains screenshots), but from the Jenkins console text it is hard to say.

So I always start with [the GitHub comment](https://github.com/liferay-frontend/liferay-portal/pull/466#issuecomment-724887699) and in this case we see `<h1>Unable to generate test report.</h1>` and `org.json.JSONException: A JSONObject text must begin with '{' at 0 [character 1 line 1]`, which to me suggests "mysterious CI failure" more than "some bug got introduced in our JS changes". So the strategy here would be to mention @john-co and @PablitoBonito for an initial assessment (they can often diagnose the issue, or direct you to `#ci` in Slack if it actually is an CI infra problem), and in the meantime re-run the tests to see if the failure is transient, which I will do presently.

**Comment by wincent** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
Thank you for the detailed reply @wincent. Hadn't you done that I'd have spent probably the entire day trying to find them. Appreciate it a lot ❤️

**Comment by wincent** [comment]
> I'd have spent probably the entire day trying to find them.

Ha, it's not forwarded yet. You still might have to spend an entire day "playing" with CI before you see the end of this 😉

**Comment by kresimir-coko** [comment]
Trying to make sense of the CI report, but I couldn't gather any actionable information from it @wincent got any insights?

**Comment by wincent** [comment]
> Trying to make sense of the CI report, but I couldn't gather any actionable information from it @wincent got any insights?

The first few failures look like CI infra just blowing up. Which is enough to make me re-trigger this to see if we get a better result. You can also ask @john-co and @PablitoBonito, our QA folks, whenever you have trouble analyzing test failures — they are the real experts on this.

**Comment by wincent** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
Okay, I know you told me about pinging them yesterday, but I didn't want to burden them if it's something I can fix. I'll be slightly more trigger-happy in the future with that. Thanks @wincent , it's kinda hard identifying CI patterns and what to do with them.

**Comment by wincent** [comment]
Yeah, I'd say be more trigger happy for sure. If anything, we have erred to far in the other direction in the past. If we overcorrect, I am sure they'll let us know. 😂

**Comment by john-co** [comment]
> Hey @wincent I'm trying to see which files/modules have SF errors in them, but I don't know the login details for the test platform:
> image

I think you're trying to access either testray or poshi reports, if that's the case I believe you can use liferay lax vpn to bypass the login credentials.

**Comment by john-co** [comment]
ci:test:bundle

**Comment by john-co** [comment]
From reviewing the results.

- A number of poshi tests are timing out waiting for SPA to finish loading after clicking save on Add Site modal.

- There's a layouts integration test that failed:

```
     [exec] > Task :apps:layout:layout-content-page-editor-web-test:testIntegration
     [exec] Excluded project dependency :test:arquillian-extension-junit-bridge for :apps:layout:layout-content-page-editor-web-test
     [exec] 
     [exec] com.liferay.layout.content.page.editor.web.internal.adaptive.media.test.LayoutAdaptiveMediaProcessorTest STANDARD_OUT
     [exec]     -- listing properties --
     [exec]     assert.logs=true
     [exec]     ci.test.timeout.time=1200000
     [exec]     company.web.id=liferay.com
     [exec]     dl.file.entry.processors.trigger.synchronously=true
     [exec]     include-and-override=test-portal-impl-ext.properties
     [exec]     journal.transformer.regex.pattern.0=beta.sample.com
     [exec]     journal.transformer.regex.replacement.0=production.sample.com
     [exec]     junit.delay.factor=1
     [exec]     portal.url=http://localhost:8080
     [exec]     user.password=********
     [exec] 
     [exec]     -- updated properties --
     [exec]     assert.logs=true
     [exec]     ci.test.timeout.time=1200000
     [exec]     company.web.id=liferay.com
     [exec]     dl.file.entry.processors.trigger.synchronously=true
     [exec]     include-and-override=test-portal-impl-ext.properties
     [exec]     journal.transformer.regex.pattern.0=beta.sample.com
     [exec]     journal.transformer.regex.replacement.0=production.sample.com
     [exec]     junit.delay.factor=1
     [exec]     portal.url=http://localhost:8080
     [exec]     user.password=********
     [exec] 
     [exec]     Loading jar:file:/opt/dev/projects/github/liferay-portal/bundles/tomcat-9.0.37/webapps/ROOT/WEB-INF/lib/portal-impl.jar!/system.properties
     [exec]     Loading file:/opt/dev/projects/github/liferay-portal/bundles/tomcat-9.0.37/webapps/ROOT/WEB-INF/classes/system-ext.properties
     [exec] 16:19:04,708 INFO  [Arquillian-Connector-Thread][BundleStartStopLogger:46] STARTED com.liferay.layout.content.page.editor.web.test_1.0.0 [1337]
     [exec] 16:19:05,282 WARN  [CM Configuration Updater (Update: pid=com.liferay.layout.content.page.editor.web.internal.configuration.FFLayoutContentPageEditorConfiguration)][LogService:99] Exception occured on JMX Notification dispatch for event [org.osgi.framework.ServiceEvent[source={com.liferay.layout.content.page.editor.web.internal.configuration.FFLayoutContentPageEditorConfiguration}={service.id=6993, service.bundleid=254, service.scope=singleton}]] 
     [exec] java.util.ConcurrentModificationException
     [exec] 	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:901)
     [exec] 	at java.util.ArrayList$Itr.next(ArrayList.java:851)
     [exec] 	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.getServiceReferences(ServiceRegistry.java:314)
     [exec] 	at org.eclipse.osgi.internal.framework.BundleContextImpl.getAllServiceReferences(BundleContextImpl.java:540)
     [exec] 	at org.apache.aries.jmx.framework.ServiceState.getServiceIds(ServiceState.java:205)
     [exec] 	at org.apache.aries.jmx.framework.ServiceState.getAttributeChangeNotification(ServiceState.java:304)
     [exec] 	at org.apache.aries.jmx.framework.ServiceState$1.serviceChanged(ServiceState.java:268)
     [exec] 	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:109)
     [exec] 	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:891)
     [exec] 	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:804)
     [exec] 	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.unregister(ServiceRegistrationImpl.java:222)
     [exec] 	at com.liferay.portal.configuration.settings.internal.ConfigurationBeanManagedService.doUpdated(ConfigurationBeanManagedService.java:122)
     [exec] 	at com.liferay.portal.configuration.settings.internal.ConfigurationBeanManagedService.updated(ConfigurationBeanManagedService.java:88)
     [exec] 	at org.apache.felix.cm.impl.helper.ManagedServiceTracker.updated(ManagedServiceTracker.java:189)
     [exec] 	at org.apache.felix.cm.impl.helper.ManagedServiceTracker.updateService(ManagedServiceTracker.java:152)
     [exec] 	at org.apache.felix.cm.impl.helper.ManagedServiceTracker.provideConfiguration(ManagedServiceTracker.java:85)
     [exec] 	at org.apache.felix.cm.impl.ConfigurationManager$UpdateConfiguration.run(ConfigurationManager.java:1391)
     [exec] 	at org.apache.felix.cm.impl.UpdateThread.run0(UpdateThread.java:138)
     [exec] 	at org.apache.felix.cm.impl.UpdateThread.run(UpdateThread.java:105)
     [exec] 	at java.lang.Thread.run(Thread.java:745)
...
...
...
    [junit] 15:58:27,959 ERROR [main][PortletPanelAppAdapterRegistry:93] bundle com.liferay.application.list.api:7.0.0 (521)[com.liferay.application.list.adapter.PortletPanelAppAdapterRegistry(1794)] : The activate method has thrown an exception 
    [junit] java.util.ConcurrentModificationException
    [junit] 	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:901)
    [junit] 	at java.util.ArrayList$Itr.next(ArrayList.java:851)
    [junit] 	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.getServiceReferences(ServiceRegistry.java:314)
    [junit] 	at org.eclipse.osgi.internal.framework.BundleContextImpl.getServiceReferences(BundleContextImpl.java:535)
    [junit] 	at org.osgi.util.tracker.ServiceTracker.getInitialReferences(ServiceTracker.java:339)
    [junit] 	at org.osgi.util.tracker.ServiceTracker.open(ServiceTracker.java:300)
    [junit] 	at org.osgi.util.tracker.ServiceTracker.open(ServiceTracker.java:264)
    [junit] 	at com.liferay.osgi.util.ServiceTrackerFactory.open(ServiceTrackerFactory.java:97)
    [junit] 	at com.liferay.application.list.adapter.PortletPanelAppAdapterRegistry.activate(PortletPanelAppAdapterRegistry.java:45)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    [junit] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [junit] 	at java.lang.reflect.Method.invoke(Method.java:498)
    [junit] 	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invokeMethod(BaseMethod.java:228)
    [junit] 	at org.apache.felix.scr.impl.inject.methods.BaseMethod.access$500(BaseMethod.java:41)
    [junit] 	at org.apache.felix.scr.impl.inject.methods.BaseMethod$Resolved.invoke(BaseMethod.java:664)
    [junit] 	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invoke(BaseMethod.java:510)
    [junit] 	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:310)
    [junit] 	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:300)
    [junit] 	at org.apache.felix.scr.impl.manager.SingleComponentManager.createImplementationObject(SingleComponentManager.java:341)
    [junit] 	at org.apache.felix.scr.impl.manager.SingleComponentManager.createComponent(SingleComponentManager.java:114)
    [junit] 	at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:983)
    [junit] 	at org.apache.felix.scr.impl.manager.SingleComponentManager.getServiceInternal(SingleComponentManager.java:956)
    [junit] 	at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:756)
    [junit] 	at org.apache.felix.scr.impl.manager.DependencyManager$SingleStaticCustomizer.addedService(DependencyManager.java:1053)
    [junit] 	at org.apache.felix.scr.impl.manager.DependencyManager$SingleStaticCustomizer.addedService(DependencyManager.java:1007)
    [junit] 	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1216)
    [junit] 	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1137)
    [junit] 	at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.trackAdding(ServiceTracker.java:944)
    [junit] 	at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.track(ServiceTracker.java:880)
    [junit] 	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:1168)
    [junit] 	at org.apache.felix.scr.impl.BundleComponentActivator$ListenerInfo.serviceChanged(BundleComponentActivator.java:125)
    [junit] 	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:109)
    [junit] 	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:891)
    [junit] 	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:804)
    [junit] 	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:127)
    [junit] 	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:228)
    [junit] 	at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:469)
    [junit] 	at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:487)
    [junit] 	at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:1004)
    [junit] 	at com.liferay.registry.internal.RegistryImpl.registerService(RegistryImpl.java:342)
    [junit] 	at com.liferay.portal.internal.servlet.MainServlet._registerPortalInitialized(MainServlet.java:1284)
    [junit] 	at com.liferay.portal.internal.servlet.MainServlet.init(MainServlet.java:443)
    [junit] 	at javax.servlet.GenericServlet.init(GenericServlet.java:158)
    [junit] 	at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1134)
    [junit] 	at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1089)
    [junit] 	at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:983)
    [junit] 	at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:4864)
    [junit] 	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5173)
    [junit] 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
    [junit] 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:717)
    [junit] 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:690)
    [junit] 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:705)
    [junit] 	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:631)
    [junit] 	at org.apache.catalina.startup.HostConfig$DeployDescriptor.run(HostConfig.java:1830)
    [junit] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    [junit] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    [junit] 	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
    [junit] 	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
    [junit] 	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:526)
    [junit] 	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:425)
    [junit] 	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1576)
    [junit] 	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:309)
    [junit] 	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
    [junit] 	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423)
    [junit] 	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366)
    [junit] 	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:936)
    [junit] 	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:841)
    [junit] 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
    [junit] 	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384)
    [junit] 	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1374)
    [junit] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    [junit] 	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
    [junit] 	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:134)
    [junit] 	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:909)
    [junit] 	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:262)
    [junit] 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
    [junit] 	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:421)
    [junit] 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
    [junit] 	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:930)
    [junit] 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
    [junit] 	at org.apache.catalina.startup.Catalina.start(Catalina.java:738)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    [junit] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [junit] 	at java.lang.reflect.Method.invoke(Method.java:498)
    [junit] 	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:342)
    [junit] 	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:473)
```
[Full console log](https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=14,label_exp=!master/450409/consoleFull)

**Comment by john-co** [comment]
Recommend to confirm if the integration failure comes up again with the 2nd round of `ci:test:relevant` and verify the SPA loading manually with the test bundle.

**Comment by john-co** [comment]
may be some [flakiness going on with the modules-integration tests](https://testray.liferay.com/home/-/testray/cases/1326330352)...but there's definitely something consistently failing with the add site modal waiting for SPA.

**Comment by kresimir-coko** [comment]
Thanks @john-co this shines some light on the real situation, gonna handle that right away!

**Comment by kresimir-coko** [comment]
I've fixed the `removeListener` issue, but I can't reproduce the 2nd problem.

For me clicking on the `Add Elements` button opens the sidebar without any errors, a problem I noticed with it was the tooltip value being `true` when changing Chrome tabs and going back to the tab where the server is active, in tha situation this is how the button looks
image

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by wincent** [comment]
> You cannot perform that action because you are not a member of the Liferay organization.

youre-fired

**Comment by kresimir-coko** [comment]
Hey @john-co I've fixed the problem in the Related section you mentioned, leftover CI errors are probably not related to this PR, can you confirm?

**Comment by john-co** [comment]
> I can reproduce this issue when running the poshi test locally. I think what's happening is that the text field is still being focused after the test clicks on close the side panel. And while the text field is still being focused, the test clicks on the add/plus button to open the side panel again but is showing the edit field side panel instead of the add Elements tab. Maybe this is something Forms team can confirm as a test fix or regression to fix? @cvanut

cc/ @cvanut, can you please take a look? The failing tests look like they're all Forms tests.

**Comment by john-co** [comment]
pinged Cleyton on slack to advise.

**Comment by john-co** [comment]
Cleyton says he will take a look tomorrow.

**Comment by jbalsas** [comment]
Hey @kresimir-coko, could you rebase this in the meantime? 

In the future, it's best to split these tasks in different commits so you can discard whatever area is giving you trouble to avoid the whole thing getting blocked over an issue in one specific area. 

If you want, you could actually rework this PR to split the changes in Forms from the rest and send 2 different PRs to see if you can get something going.

**Comment by kresimir-coko** [comment]
@jbalsas Sure, I'm fixing conflicts as we speak, so I'll split Forms away and send 2 separate PRs

**Comment by cvanut** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by john-co** [comment]
Here's the differences I'm seeing when running poshi test from this PR test bundle:
Peek 2020-11-17 12-08

vs. a bundle from master:
Expected

After clicking the Close (x) icon from the side panel, the Text Field is still highlighted (border focused). And then when the test clicks on the Plus button again, the side panel is supposed to show Elements tab to add another element, but instead has Basics tab which edits the Text Field.

You can see from the video from the master bundle that the Text Field is not highligted (border focused). And then when it clicks on Plus button, it correctly shows the side panel with Elements tab in order to add another element.

There's no difference between Chrome 86 and 65. I was able to reproduce the issue on both chrome versions running the poshi test locally. However, I cannot reproduce the issue manually.  With the current tools we have I can only think of including a refresh page that could be a possible test fix. Could this be resolved on the poshi side @kenjiheigel? Maybe there's some dependent behavior with dom.delegate?

**Comment by john-co** [comment]
I'm unable to reproduce this behavior manually with my theory about the Text Field being highlighted...so my guess is it's something on poshi side that needs a fix or something deeper:
Unreproducible

**Comment by jbalsas** [comment]
Going to close this since we're apparently trying to split it into smaller PRs like https://github.com/liferay-frontend/liferay-portal/pull/466 😉

**Pull request closed without merging**